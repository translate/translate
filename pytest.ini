[pytest]
norecursedirs = .git build docs
