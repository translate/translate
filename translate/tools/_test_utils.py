import sys

import pytest

requires_py38_mark = pytest.mark.skipif(
    sys.version_info < (3, 8), reason="Requires python 3.8"
)

test_po_files = [
    "tests/cli/data/test_pocount_po_csv/one.po",
    "tests/cli/data/test_pocount_po_file/one.po",
    "tests/cli/data/test_pocount_po_fuzzy/one.po",
]
