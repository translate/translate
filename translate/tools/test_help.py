import subprocess

import pytest

from ._test_utils import requires_py38_mark


@requires_py38_mark
@pytest.mark.parametrize(
    "command",
    [
        "build_tmdb",
        "phppo2pypo",
        "poclean",
        "pocompile",
        "poconflicts",
        "pocount",
        "podebug",
        "pogrep",
        "pomerge",
        "porestructure",
        "posegment",
        "poswap",
        "poterminology",
        "pretranslate",
        "pydiff",
        "pypo2phppo",
    ],
)
def test_help(command: str, snapshot):
    stdout = subprocess.check_output([command, "--help"], text=True)

    assert stdout == snapshot
