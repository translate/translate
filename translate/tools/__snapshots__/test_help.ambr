# serializer version: 1
# name: test_help[build_tmdb]
  '''
  usage: build_tmdb [-h] [-d TMDB_FILE] [-s SOURCE_LANG] -t TARGET_LANG
                    input files [input files ...]
  
  positional arguments:
    input files
  
  options:
    -h, --help            show this help message and exit
    -d TMDB_FILE, --tmdb TMDB_FILE
                          translation memory database file (default: tm.db)
    -s SOURCE_LANG, --import-source-lang SOURCE_LANG
                          source language of translation files (default: en)
    -t TARGET_LANG, --import-target-lang TARGET_LANG
                          target language of translation files
  
  '''
# ---
# name: test_help[phppo2pypo]
  '''
  Usage: phppo2pypo [--version] [-h|--help] [--manpage] [--progress PROGRESS] [--errorlevel ERRORLEVEL] [-i|--input] INPUT [-x|--exclude EXCLUDE] [-o|--output] OUTPUT [-S|--timestamp]
  
  Convert PHP format .po files to Python format .po files.
  
  Options:
    --version             show program's version number and exit
    -h, --help            show this help message and exit
    --manpage             output a manpage based on the help
    --progress=PROGRESS   show progress as: dots, none, bar, names, verbose
    --errorlevel=ERRORLEVEL
                          show errorlevel as: none, message, exception,
                          traceback
    -i INPUT, --input=INPUT
                          read from INPUT in po, pot formats
    -x EXCLUDE, --exclude=EXCLUDE
                          exclude names matching EXCLUDE from input paths
    -o OUTPUT, --output=OUTPUT
                          write to OUTPUT in po, pot formats
    -S, --timestamp       skip conversion if the output file has newer timestamp
  
  '''
# ---
# name: test_help[poclean]
  '''
  Usage: poclean [--version] [-h|--help] [--manpage] [--progress PROGRESS] [--errorlevel ERRORLEVEL] [-i|--input] INPUT [-x|--exclude EXCLUDE] [-o|--output] OUTPUT [-S|--timestamp]
  
  Produces a clean file from an unclean file (Trados/Wordfast) by stripping out
  the tw4win indicators.  This does not convert an RTF file to PO/XLIFF, but
  produces the target file with only the target text in from a text version of
  the RTF.
  
  Options:
    --version             show program's version number and exit
    -h, --help            show this help message and exit
    --manpage             output a manpage based on the help
    --progress=PROGRESS   show progress as: dots, none, bar, names, verbose
    --errorlevel=ERRORLEVEL
                          show errorlevel as: none, message, exception,
                          traceback
    -i INPUT, --input=INPUT
                          read from INPUT in po, pot, xlf, xliff formats
    -x EXCLUDE, --exclude=EXCLUDE
                          exclude names matching EXCLUDE from input paths
    -o OUTPUT, --output=OUTPUT
                          write to OUTPUT in po, pot, xlf, xliff formats
    -S, --timestamp       skip conversion if the output file has newer timestamp
  
  '''
# ---
# name: test_help[pocompile]
  '''
  Usage: pocompile [--version] [-h|--help] [--manpage] [--progress PROGRESS] [--errorlevel ERRORLEVEL] [-i|--input] INPUT [-x|--exclude EXCLUDE] [-o|--output] OUTPUT [-S|--timestamp]
  
  Compile XLIFF and Gettext PO localization files into Gettext MO (Machine
  Object) files.  See: http://docs.translatehouse.org/projects/translate-
  toolkit/en/latest/commands/pocompile.html for examples and usage instructions.
  
  Options:
    --version             show program's version number and exit
    -h, --help            show this help message and exit
    --manpage             output a manpage based on the help
    --progress=PROGRESS   show progress as: dots, none, bar, names, verbose
    --errorlevel=ERRORLEVEL
                          show errorlevel as: none, message, exception,
                          traceback
    -i INPUT, --input=INPUT
                          read from INPUT in po, pot, xlf, xliff formats
    -x EXCLUDE, --exclude=EXCLUDE
                          exclude names matching EXCLUDE from input paths
    -o OUTPUT, --output=OUTPUT
                          write to OUTPUT in mo format
    -S, --timestamp       skip conversion if the output file has newer timestamp
    --fuzzy               use translations marked fuzzy
    --nofuzzy             don't use translations marked fuzzy (default)
  
  '''
# ---
# name: test_help[poconflicts]
  '''
  Usage: poconflicts [--version] [-h|--help] [--manpage] [--progress PROGRESS] [--errorlevel ERRORLEVEL] [-i|--input] INPUT [-x|--exclude EXCLUDE] [-o|--output] OUTPUT [-I|--ignore-case] [-v|--invert] [--accelerator ACCELERATORS]
    input directory is searched for PO files, PO files with name of conflicting string are output in output directory
  
  Conflict finder for Gettext PO localization files.  See:
  http://docs.translatehouse.org/projects/translate-
  toolkit/en/latest/commands/poconflicts.html for examples and usage
  instructions.
  
  Options:
    --version             show program's version number and exit
    -h, --help            show this help message and exit
    --manpage             output a manpage based on the help
    --progress=PROGRESS   show progress as: dots, none, bar, names, verbose
    --errorlevel=ERRORLEVEL
                          show errorlevel as: none, message, exception,
                          traceback
    -i INPUT, --input=INPUT
                          read from INPUT in po format
    -x EXCLUDE, --exclude=EXCLUDE
                          exclude names matching EXCLUDE from input paths
    -o OUTPUT, --output=OUTPUT
                          write to OUTPUT in po format
    -I, --ignore-case     ignore case distinctions
    -v, --invert          invert the conflicts thus extracting conflicting
                          destination words
    --accelerator=ACCELERATORS
                          ignores the given accelerator characters when matching
  
  '''
# ---
# name: test_help[pocount]
  '''
  usage: pocount [-h] [--incomplete]
                 [--full | --csv | --short | --short-strings | --short-words]
                 [--no-color]
                 files [files ...]
  
  positional arguments:
    files
  
  options:
    -h, --help       show this help message and exit
    --incomplete     skip 100% translated files.
  
  Output format:
    --full           (default) statistics in full, verbose format
    --csv            statistics in CSV format
    --short          same as --short-strings
    --short-strings  statistics of strings in short format - one line per file
    --short-words    statistics of words in short format - one line per file
    --no-color       show output without color
  
  '''
# ---
# name: test_help[podebug]
  '''
  Usage: podebug [--version] [-h|--help] [--manpage] [--progress PROGRESS] [--errorlevel ERRORLEVEL] [-i|--input] INPUT [-x|--exclude EXCLUDE] [-o|--output] OUTPUT [-S|--timestamp]
  
  Insert debug messages into XLIFF and Gettext PO localization files.  See:
  http://docs.translatehouse.org/projects/translate-
  toolkit/en/latest/commands/podebug.html for examples and usage instructions.
  
  Options:
    --version             show program's version number and exit
    -h, --help            show this help message and exit
    --manpage             output a manpage based on the help
    --progress=PROGRESS   show progress as: dots, none, bar, names, verbose
    --errorlevel=ERRORLEVEL
                          show errorlevel as: none, message, exception,
                          traceback
    -i INPUT, --input=INPUT
                          read from INPUT in po, pot, pot, tmx, xlf, xliff
                          formats
    -x EXCLUDE, --exclude=EXCLUDE
                          exclude names matching EXCLUDE from input paths
    -o OUTPUT, --output=OUTPUT
                          write to OUTPUT in po, pot, tmx, xlf, xliff formats
    -S, --timestamp       skip conversion if the output file has newer timestamp
    -f FORMAT, --format=FORMAT
                          specify format string
    --rewrite=STYLE       the translation rewrite style: blank, bracket, chef,
                          classified, en, flipped, unicode, xxx
    --ignore=APPLICATION  apply tagging ignore rules for the given application:
                          gtk, kde, libreoffice, mozilla, openoffice
    --preserveplaceholders
                          attempt to exclude characters that are part of
                          placeholders when performing character-level rewrites
                          so that consuming applications can still use the
                          placeholders to generate final output
  
  '''
# ---
# name: test_help[pogrep]
  '''
  Usage: pogrep searchstring [--version] [-h|--help] [--manpage] [--progress PROGRESS] [--errorlevel ERRORLEVEL] [-i|--input] INPUT [-x|--exclude EXCLUDE] [-o|--output] OUTPUT [--search SEARCHPARTS] [-I|--ignore-case] [-e|--regexp] [-v|--invert-match] [--accelerator ACCELERATOR] [-k|--keep-translations]
  
  Grep XLIFF, Gettext PO and TMX localization files.  Matches are output to
  snippet files of the same type which can then be reviewed and later merged
  using :doc:`pomerge </commands/pomerge>`.  See:
  http://docs.translatehouse.org/projects/translate-
  toolkit/en/latest/commands/pogrep.html for examples and usage instructions.
  
  Options:
    --version             show program's version number and exit
    -h, --help            show this help message and exit
    --manpage             output a manpage based on the help
    --progress=PROGRESS   show progress as: dots, none, bar, names, verbose
    --errorlevel=ERRORLEVEL
                          show errorlevel as: none, message, exception,
                          traceback
    -i INPUT, --input=INPUT
                          read from INPUT in gmo, mo, po, pot, tmx, xlf, xlff,
                          xliff formats
    -x EXCLUDE, --exclude=EXCLUDE
                          exclude names matching EXCLUDE from input paths
    -o OUTPUT, --output=OUTPUT
                          write to OUTPUT in gmo, mo, po, pot, tmx, xlf, xlff,
                          xliff formats
    --search=SEARCHPARTS  searches the given parts (source, target, notes and
                          locations)
    -I, --ignore-case     ignore case distinctions
    -e, --regexp          use regular expression matching
    -v, --invert-match    select non-matching lines
    --accelerator=ACCELERATOR
                          ignores the given accelerator when matching
    -k, --keep-translations
                          always extract units with translations
  
  '''
# ---
# name: test_help[pomerge]
  '''
  Usage: pomerge [--version] [-h|--help] [--manpage] [--progress PROGRESS] [--errorlevel ERRORLEVEL] [-i|--input] INPUT [-x|--exclude EXCLUDE] [-o|--output] OUTPUT [-t|--template TEMPLATE] [-S|--timestamp]
  
  Merges XLIFF and Gettext PO localization files.  Snippet file produced by e.g.
  :doc:`pogrep </commands/pogrep>` and updated by a translator can be merged
  back into the original files.  See:
  http://docs.translatehouse.org/projects/translate-
  toolkit/en/latest/commands/pomerge.html for examples and usage instructions.
  
  Options:
    --version             show program's version number and exit
    -h, --help            show this help message and exit
    --manpage             output a manpage based on the help
    --progress=PROGRESS   show progress as: dots, none, bar, names, verbose
    --errorlevel=ERRORLEVEL
                          show errorlevel as: none, message, exception,
                          traceback
    -i INPUT, --input=INPUT
                          read from INPUT in po, pot, pot, xlf, xliff formats
    -x EXCLUDE, --exclude=EXCLUDE
                          exclude names matching EXCLUDE from input paths
    -o OUTPUT, --output=OUTPUT
                          write to OUTPUT in po, pot, pot, xlf, xliff formats
    -t TEMPLATE, --template=TEMPLATE
                          read from TEMPLATE in po, pot, pot, xlf, xliff formats
    -S, --timestamp       skip conversion if the output file has newer timestamp
    --mergeblanks=MERGEBLANKS
                          whether to overwrite existing translations with blank
                          translations (yes/no). Default is yes.
    --mergefuzzy=MERGEFUZZY
                          whether to consider fuzzy translations from input
                          (yes/no). Default is yes.
    --mergecomments=MERGECOMMENTS
                          whether to merge comments as well as translations
                          (yes/no). Default is yes.
  
  '''
# ---
# name: test_help[porestructure]
  '''
  Usage: porestructure [--version] [-h|--help] [--manpage] [--progress PROGRESS] [--errorlevel ERRORLEVEL] [-i|--input] INPUT [-x|--exclude EXCLUDE] [-o|--output] OUTPUT
    input directory is searched for PO files with (poconflicts) comments, all entries are written to files in a directory structure for pomerge
  
  Restructure Gettxt PO files produced by :doc:`poconflicts
  </commands/poconflicts>` into the original directory tree for merging using
  :doc:`pomerge </commands/pomerge>`.  See:
  http://docs.translatehouse.org/projects/translate-
  toolkit/en/latest/commands/pomerge.html for examples and usage instructions.
  
  Options:
    --version             show program's version number and exit
    -h, --help            show this help message and exit
    --manpage             output a manpage based on the help
    --progress=PROGRESS   show progress as: dots, none, bar, names, verbose
    --errorlevel=ERRORLEVEL
                          show errorlevel as: none, message, exception,
                          traceback
    -i INPUT, --input=INPUT
                          read from INPUT in po format
    -x EXCLUDE, --exclude=EXCLUDE
                          exclude names matching EXCLUDE from input paths
    -o OUTPUT, --output=OUTPUT
                          write to OUTPUT in po format
  
  '''
# ---
# name: test_help[posegment]
  '''
  Usage: posegment [--version] [-h|--help] [--manpage] [--progress PROGRESS] [--errorlevel ERRORLEVEL] [-i|--input] INPUT [-x|--exclude EXCLUDE] [-o|--output] OUTPUT [-S|--timestamp] [-P|--pot]
  
  Segment Gettext PO, XLIFF and TMX localization files at the sentence level.
  See: http://docs.translatehouse.org/projects/translate-
  toolkit/en/latest/commands/posegment.html for examples and usage instructions.
  
  Options:
    --version             show program's version number and exit
    -h, --help            show this help message and exit
    --manpage             output a manpage based on the help
    --progress=PROGRESS   show progress as: dots, none, bar, names, verbose
    --errorlevel=ERRORLEVEL
                          show errorlevel as: none, message, exception,
                          traceback
    -i INPUT, --input=INPUT
                          read from INPUT in po, pot, tmx, xlf, xliff formats
    -x EXCLUDE, --exclude=EXCLUDE
                          exclude names matching EXCLUDE from input paths
    -o OUTPUT, --output=OUTPUT
                          write to OUTPUT in po, pot, tmx, xlf, xliff formats
    -S, --timestamp       skip conversion if the output file has newer timestamp
    -P, --pot             output PO Templates (.pot) rather than PO files (.po)
    -l LANG, --language=LANG
                          the target language code
    --source-language=LANG
                          the source language code (default 'en')
    --keepspaces          Disable automatic stripping of whitespace
    --only-aligned        Removes units where sentence number does not
                          correspond
  
  '''
# ---
# name: test_help[poswap]
  '''
  Usage: poswap [--version] [-h|--help] [--manpage] [--progress PROGRESS] [--errorlevel ERRORLEVEL] [-i|--input] INPUT [-x|--exclude EXCLUDE] [-o|--output] OUTPUT [-t|--template TEMPLATE] [-S|--timestamp]
  
  Builds a new translation file with the target of the input language as source
  language.  .. note:: Ensure that the two po files correspond 100% to the same
  pot file before using    this.  To translate Kurdish (ku) through French::
  poswap -i fr/ -t ku -o fr-ku  To convert the fr-ku files back to en-ku::
  poswap --reverse -i fr/ -t fr-ku -o en-ku  See:
  http://docs.translatehouse.org/projects/translate-
  toolkit/en/latest/commands/poswap.html for examples and usage instructions.
  
  Options:
    --version             show program's version number and exit
    -h, --help            show this help message and exit
    --manpage             output a manpage based on the help
    --progress=PROGRESS   show progress as: dots, none, bar, names, verbose
    --errorlevel=ERRORLEVEL
                          show errorlevel as: none, message, exception,
                          traceback
    -i INPUT, --input=INPUT
                          read from INPUT in po, pot formats
    -x EXCLUDE, --exclude=EXCLUDE
                          exclude names matching EXCLUDE from input paths
    -o OUTPUT, --output=OUTPUT
                          write to OUTPUT in po, pot formats
    -t TEMPLATE, --template=TEMPLATE
                          read from TEMPLATE in po, pot, pot formats
    -S, --timestamp       skip conversion if the output file has newer timestamp
    --reverse             reverse the process of intermediate language
                          conversion
  
  '''
# ---
# name: test_help[poterminology]
  '''
  Usage: poterminology [--version] [-h|--help] [--manpage] [--progress PROGRESS] [--errorlevel ERRORLEVEL] [-i|--input] INPUT [-x|--exclude EXCLUDE] [-o|--output] OUTPUT [-u|--update UPDATEFILE] [-S|--stopword-list STOPFILE] [-F|--fold-titlecase] [-C|--preserve-case] [-I|--ignore-case] [--accelerator ACCELERATORS] [-t|--term-words LENGTH] [--nonstop-needed MIN] [--inputs-needed MIN] [--fullmsg-needed MIN] [--substr-needed MIN] [--locs-needed MIN] [--sort ORDER] [--source-language LANG] [-v|--invert]
    input directory is searched for PO files, terminology PO file is output file
  
  Create a terminology file by reading a set of .po or .pot files to produce a
  pootle-terminology.pot.  See:
  http://docs.translatehouse.org/projects/translate-
  toolkit/en/latest/commands/poterminology.html for examples and usage
  instructions.
  
  Options:
    --version             show program's version number and exit
    -h, --help            show this help message and exit
    --manpage             output a manpage based on the help
    --progress=PROGRESS   show progress as: dots, none, bar, names, verbose
    --errorlevel=ERRORLEVEL
                          show errorlevel as: none, message, exception,
                          traceback
    -i INPUT, --input=INPUT
                          read from INPUT in po, pot formats
    -x EXCLUDE, --exclude=EXCLUDE
                          exclude names matching EXCLUDE from input paths
    -o OUTPUT, --output=OUTPUT
                          write to OUTPUT in po, pot formats
    -u UPDATEFILE, --update=UPDATEFILE
                          update terminology in UPDATEFILE
    -S STOPFILE, --stopword-list=STOPFILE
                          read stopword (term exclusion) list from STOPFILE
                          (default
                          /home/serg/src/translate/.venv/lib/python3.10/site-
                          packages/translate/misc/../share/stoplist-en)
    -F, --fold-titlecase  fold "Title Case" to lowercase (default)
    -C, --preserve-case   preserve all uppercase/lowercase
    -I, --ignore-case     make all terms lowercase
    --accelerator=ACCELERATORS
                          ignore the given accelerator characters when matching
    -t LENGTH, --term-words=LENGTH
                          generate terms of up to LENGTH words (default 3)
    --nonstop-needed=MIN  omit terms with less than MIN nonstop words (default
                          1)
    --inputs-needed=MIN   omit terms appearing in less than MIN input files
                          (default 2, or 1 if only one input file)
    --fullmsg-needed=MIN  omit full message terms appearing in less than MIN
                          different messages (default 1)
    --substr-needed=MIN   omit substring-only terms appearing in less than MIN
                          different messages (default 2)
    --locs-needed=MIN     omit terms appearing in less than MIN different
                          original source files (default 2)
    --sort=ORDER          output sort order(s): frequency, dictionary, length
                          (may repeat option, default is all in above order)
    --source-language=LANG
                          the source language code (default 'en')
    -v, --invert          invert the source and target languages for terminology
  
  '''
# ---
# name: test_help[pretranslate]
  '''
  Usage: pretranslate [--version] [-h|--help] [--manpage] [--progress PROGRESS] [--errorlevel ERRORLEVEL] [-i|--input] INPUT [-x|--exclude EXCLUDE] [-o|--output] OUTPUT [-t|--template TEMPLATE] [-S|--timestamp]
  
  Fill localization files with suggested translations based on translation
  memory and existing translations.  See:
  http://docs.translatehouse.org/projects/translate-
  toolkit/en/latest/commands/pretranslate.html for examples and usage
  instructions.
  
  Options:
    --version             show program's version number and exit
    -h, --help            show this help message and exit
    --manpage             output a manpage based on the help
    --progress=PROGRESS   show progress as: dots, none, bar, names, verbose
    --errorlevel=ERRORLEVEL
                          show errorlevel as: none, message, exception,
                          traceback
    -i INPUT, --input=INPUT
                          read from INPUT in po, pot, pot, xlf, xliff formats
    -x EXCLUDE, --exclude=EXCLUDE
                          exclude names matching EXCLUDE from input paths
    -o OUTPUT, --output=OUTPUT
                          write to OUTPUT in po, pot, xlf, xliff formats
    -t TEMPLATE, --template=TEMPLATE
                          read from TEMPLATE in po, pot, xlf, xliff formats
    -S, --timestamp       skip conversion if the output file has newer timestamp
    --tm=TM               The file to use as translation memory when fuzzy
                          matching
    -s MIN_SIMILARITY, --similarity=MIN_SIMILARITY
                          The minimum similarity for inclusion (default: 75%)
    --nofuzzymatching     Disable fuzzy matching
  
  '''
# ---
# name: test_help[pydiff]
  '''
  usage: pydiff [-h] [-i] [-U NUM] [-r] [-N] [--unidirectional-new-file] [-s]
                [-x PAT] [--fromcontains TEXT] [--tocontains TEXT]
                [--contains TEXT] [-I] [--accelerator ACCELERATORS]
                fromfile tofile
  
  positional arguments:
    fromfile
    tofile
  
  options:
    -h, --help            show this help message and exit
    -i, --ignore-case     Ignore case differences in file contents.
    -U NUM, --unified NUM
                          Output NUM (default 3) lines of unified context
    -r, --recursive       Recursively compare any subdirectories found.
    -N, --new-file        Treat absent files as empty.
    --unidirectional-new-file
                          Treat absent first files as empty.
    -s, --report-identical-files
                          Report when two files are the same.
    -x PAT, --exclude PAT
                          Exclude files that match PAT.
    --fromcontains TEXT   Only show changes where fromfile contains TEXT
    --tocontains TEXT     Only show changes where tofile contains TEXT
    --contains TEXT       Only show changes where fromfile or tofile contains
                          TEXT
    -I, --ignore-case-contains
                          Ignore case differences when matching any of the
                          changes
    --accelerator ACCELERATORS
                          ignores the given accelerator characters when matching
  
  '''
# ---
# name: test_help[pypo2phppo]
  '''
  Usage: pypo2phppo [--version] [-h|--help] [--manpage] [--progress PROGRESS] [--errorlevel ERRORLEVEL] [-i|--input] INPUT [-x|--exclude EXCLUDE] [-o|--output] OUTPUT [-S|--timestamp]
  
  Convert Python format .po files to PHP format .po files.
  
  Options:
    --version             show program's version number and exit
    -h, --help            show this help message and exit
    --manpage             output a manpage based on the help
    --progress=PROGRESS   show progress as: dots, none, bar, names, verbose
    --errorlevel=ERRORLEVEL
                          show errorlevel as: none, message, exception,
                          traceback
    -i INPUT, --input=INPUT
                          read from INPUT in po, pot formats
    -x EXCLUDE, --exclude=EXCLUDE
                          exclude names matching EXCLUDE from input paths
    -o OUTPUT, --output=OUTPUT
                          write to OUTPUT in po, pot formats
    -S, --timestamp       skip conversion if the output file has newer timestamp
  
  '''
# ---
